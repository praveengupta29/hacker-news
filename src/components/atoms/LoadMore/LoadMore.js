import React from 'react';
import PropTypes from 'prop-types';

import Anchor from '../Anchor';
import './LoadMore.scss';

const isVisible = (currentPage, totalPages) => {
  return !!(totalPages && currentPage !== totalPages);
};

const LoadMore = ({ currentPage, fetchFeedListData, totalPages }) => {
  const cp = currentPage + 1;
  return (
    isVisible(cp, totalPages) && (
      <div className="load-more sm-text-center md-text-initial">
        <Anchor
          handleLinkClick={e => {
            e.preventDefault();
            fetchFeedListData({ page: cp });
          }}
          className="load-more-link"
        >
          More
        </Anchor>
      </div>
    )
  );
};

LoadMore.propTypes = {
  currentPage: PropTypes.number.isRequired,
  fetchFeedListData: PropTypes.func.isRequired,
  totalPages: PropTypes.number.isRequired,
};

export default LoadMore;
