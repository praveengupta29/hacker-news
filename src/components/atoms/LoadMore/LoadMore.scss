@import '../../../styles/utils/colors';
@import '../../../styles/utils/breakpoints';

.load-more {
  margin-top: 16px;

  .load-more-link {
    color: $orange;
    font-size: 14px;
    @include tablet {
      margin-left: 141px;
    }
    @include desktop {
      margin-left: 165px;
    }
  }
}
