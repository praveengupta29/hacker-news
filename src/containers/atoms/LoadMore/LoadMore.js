import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import {
  selectTotalPageData,
  selectCurrentPageData,
} from '../../organisms/FeedList/selectors';
import { fetchFeedListData } from '../../organisms/FeedList/actions';

import LoadMore from '../../../components/atoms/LoadMore';

export const mapStateToProps = createStructuredSelector({
  currentPage: selectCurrentPageData,
  totalPages: selectTotalPageData,
});

export const mapDispatchToProps = dispatch => {
  return {
    fetchFeedListData: params => {
      dispatch(fetchFeedListData(params));
    },
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(LoadMore);
