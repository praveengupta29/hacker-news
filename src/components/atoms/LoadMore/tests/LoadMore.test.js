import React from 'react';
import { shallow } from 'enzyme';
import LoadMore from '../LoadMore';

describe('LoadMore component', () => {
  const props = {
    currentPage: 1,
    fetchFeedListData: () => {},
    totalPages: 3,
  };

  it('should render correctly and match snapshot', () => {
    const component = shallow(<LoadMore {...props} />);
    expect(component).toMatchSnapshot();
  });
});
