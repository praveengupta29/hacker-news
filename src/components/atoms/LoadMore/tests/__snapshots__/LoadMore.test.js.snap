// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadMore component should render correctly and match snapshot 1`] = `
<div
  className="load-more sm-text-center md-text-initial"
>
  <Anchor
    className="load-more-link"
    handleLinkClick={[Function]}
    target="_self"
    to=""
  >
    More
  </Anchor>
</div>
`;
